lib/libc/minimal/include/sys/types.h

```diff
@@ -32,27 +32,8 @@ typedef __SIZE_TYPE__ ssize_t;
 
 #if !defined(__off_t_defined)
 #define __off_t_defined
-
-#if defined(__i386) || defined(__x86_64) || defined(__ARC64__)
-typedef long int off_t; /* "long" works for all of i386, X32 and true 64 bit */
-#elif defined(__ARM_ARCH)
-typedef int off_t;
-#elif defined(__arc__)
-typedef int off_t;
-#elif defined(__NIOS2__)
-typedef int off_t;
-#elif defined(__riscv)
-typedef int off_t;
-#elif defined(__XTENSA__)
-typedef int off_t;
-#elif defined(__sparc__)
-typedef int off_t;
-#elif defined(__mips)
-typedef int off_t;
-#else
-#error "The minimal libc library does not recognize the architecture!\n"
-#endif
-
+/* off_t is defined such that it matches the size of a pointer */
+typedef __INTPTR_TYPE__ off_t;
```

Comment:
I definitely agree that `off_t` should not depend on the architecture. Do you want it to vary based on the size of machine registers though? That means moving from a 32-bit target to a 64-bit target changes the potential value range of a huge number of APIs. In addition, you're making a breaking API change on all many 64-bit platforms here, including risc-v and aarch64. That change will make the minimal libc more consistent with picolibc and newlib, which seems good in some ways.

Secondly, is `__INTPTR_TYPE__` the right abstract naming for register sizes? The history of POSIX uses `off_t` in places where the old API used `long`. That's because 16-bit systems needed (at least) 32-bits for file sizes; pointers were 16 bits, but `long` was 32-bits. 

If you want to use 32-bits on ILP32 systems and 64-bits on LP64 systems, I'd suggest `long`. If, instead, you want to use 64-bits on all systems to ensure consistency across register sizes, then using `int64_t` would be a better choice. Either of these is a breaking ABI/API change on some systems; `long` breaks many 64-bit targets, `int64_t` breaks those as well as all 32-bit targets.

Reply:
FYI, that commit is from https://github.com/zephyrproject-rtos/zephyr/pull/47612